### PR TITLE
Futurize for Py3 

### PR DESCRIPTION
**Pull request by MrSaints**
As per the suggestion https://github.com/uber/jaeger-client-python/pull/43#issuecomment-308895785, this PR runs `futurize` against the source with small fixes to ensure the Python 2 tests continues to pass. I thought I make this PR to get Python 3 support moving (even if it is in the smallest way possible).

### CODE REVIEW

**Comment by yurishkuro** [comment]
Thanks for this PR, @MrSaints!

This change looks good, mostly. The two concerns I have are:
  1. The use of `old_div` seems unnecessary. There aren't that many occurrences, can we replace them with `/` for floats and `//` for ints? It is usually apparent from the context which numbers we're dealing with.
  1. The change from `iteritems()/iterkeys()` to `items()/keys()` is not good, since in Py2 the latter will cause creation of new arrays. Can we replace all instances with respective proxy methods from `six`?

And there are some lint errors in the build:
```
flake8 jaeger_client crossdock tests
jaeger_client/codecs.py:25:1: E402 module level import not at top of file
jaeger_client/codecs.py:26:1: E402 module level import not at top of file
jaeger_client/codecs.py:27:1: E402 module level import not at top of file
jaeger_client/codecs.py:27:22: E401 multiple imports on one line
jaeger_client/codecs.py:29:1: E402 module level import not at top of file
jaeger_client/codecs.py:33:1: E402 module level import not at top of file
jaeger_client/codecs.py:38:1: E402 module level import not at top of file
jaeger_client/metrics.py:28:1: E302 expected 2 blank lines, found 0
make: *** [lint] Error 1
```

**Comment by MrSaints** [comment]
Cheers for the quick feedback @yurishkuro. I've addressed them in the new commits.

File: jaeger_client/sampler.py [hunk 1]
```diff
@@ -19,9 +19,12 @@
 # THE SOFTWARE.
 
 from __future__ import absolute_import
+from __future__ import division
```
Comment (yurishkuro) [inline jaeger_client/sampler.py]: Q: is this import necessary?

File: tests/test_sampler.py [hunk 2]
```diff
@@ -198,7 +201,7 @@ def test_adaptive_sampler():
     sampled, tags = sampler.is_sampled(MAX_INT-10, "new_op")
     assert sampled
     assert tags == get_tags('probabilistic', 0.51)
-    sampled, tags = sampler.is_sampled(MAX_INT+(MAX_INT/4), "new_op")
+    sampled, tags = sampler.is_sampled(MAX_INT+(old_div(MAX_INT,4)), "new_op")
```
Comment (yurishkuro) [inline tests/test_sampler.py]: can we change back to `/`?

File: tests/test_sampler.py [hunk 3]
```diff
@@ -207,7 +210,7 @@ def test_adaptive_sampler():
     sampled, tags = sampler.is_sampled(MAX_INT-10, "new_op_2")
     assert sampled
     assert tags == get_tags('probabilistic', 0.51)
-    sampled, _ = sampler.is_sampled(MAX_INT+(MAX_INT/4), "new_op_2")
+    sampled, _ = sampler.is_sampled(MAX_INT+(old_div(MAX_INT,4)), "new_op_2")
```
Comment (yurishkuro) [inline tests/test_sampler.py]: can we change back to `/`?

File: tests/test_sampler.py [hunk 4]
```diff
@@ -18,6 +19,8 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
+from builtins import range
+from past.utils import old_div
```
Comment (yurishkuro) [inline tests/test_sampler.py]: can we remove this?

File: tests/test_sampler.py [hunk 5]
```diff
@@ -167,7 +170,7 @@ def test_guaranteed_throughput_probabilistic_sampler():
     sampled, tags = sampler.is_sampled(MAX_INT-10)
     assert sampled
     assert tags == get_tags('probabilistic', 0.51)
-    sampled, tags = sampler.is_sampled(MAX_INT+(MAX_INT/4))
+    sampled, tags = sampler.is_sampled(MAX_INT+(old_div(MAX_INT,4)))
```
Comment (yurishkuro) [inline tests/test_sampler.py]: can we use `/`?

File: jaeger_client/sampler.py (hunk 1 above)
Comment (MrSaints) [inline jaeger_client/sampler.py]: I'm not too sure. Seems like it.

```
The future division statement, spelled from __future__ import division , will change the / operator to mean true division throughout the module.
```

File: jaeger_client/thrift.py [hunk 6]
```diff
@@ -117,7 +118,7 @@ def timestamp_micros(ts):
     :param ts:
     :return:
     """
-    return long(ts * 1000000)
+    return int(ts * 1000000)
```
Comment (MrSaints) [inline jaeger_client/thrift.py]: We can probably switch between `int`, and `long` using `six`.

File: tests/test_sampler.py (hunk 4 above)
Comment (MrSaints) [inline tests/test_sampler.py]: Done :)

**Comment by MrSaints** [comment]
I've made the changes @yurishkuro, but I'm not too sure what's going on with one of the Travis `COVER=1` test: `INTERNALERROR> RuntimeError: dictionary changed size during iteration`

**Comment by black-adder** [comment]
The code coverage issue looks related to: https://bitbucket.org/ned/coveragepy/issues/581/44b1-44-breaking-in-ci we probably want to use an older version of coverage until it's fixed

File: jaeger_client/thrift.py (hunk 6 above)
Comment (black-adder) [inline jaeger_client/thrift.py]: do we need to make that change for this PR?

Comment (MrSaints) [inline jaeger_client/thrift.py]: I've made that change, I'm unsure if the `int` behaves as expected for py3, but we can do that in a different PR.

**Comment by yurishkuro** [comment]
> `long_fn = int if six.PY3 else long`

@MrSaints a couple questions:
  * Will ^^ actually parse/compile with Py3?  I thought 3.x decommissioned `long`.
  * from reading the [PEP 237](https://www.python.org/dev/peps/pep-0237/) and [SO question](https://stackoverflow.com/questions/2104884/how-does-python-manage-int-and-long) it seems the overflow on ints has been eliminated in favor of automatic upgrading of the value from a machine word to a malloc-ed memory, making this transparent to the end user. I don't know if we can find a 32bit machine to test this, any thoughts?

Even on 64bit machine the automatic type conversion can be seen:
```python
>>> import sys;
>>> sys.maxint
9223372036854775807
>>> type(sys.maxint)
<type 'int'>
>>> type(sys.maxint + 1)
<type 'long'>
>>> type(sys.maxint * 100)
<type 'long'>
>>> (type(sys.maxint * 100), sys.maxint * 100)
(<type 'long'>, 922337203685477580700L)
```

i.e. an `int` value times 100 is auto-converted to `long`.

**Comment by MrSaints** [comment]
(1) Whoops @yurishkuro. I've cleaned it up. 
(2) We can do more vigorous testing with Python 3. I believe it should be fine if it does not change the existing behaviour in Python 2.

**Comment by yurishkuro** [comment]
Attempting to fix the coverage issue here #61

**Comment by yurishkuro** [comment]
@MrSaints can you rebase or merge from master to pick up #61?

**Comment by MrSaints** [comment]
@yurishkuro Cheers for the fix. And rebased.

**Comment by MrSaints** [comment]
We good to merge?

**Comment by MrSaints** [comment]
Cheers for all the feedback, and for pushing this out @yurishkuro @black-adder

**Comment by Groxx** [comment]
Does this `install_aliases()` call taint the entire environment that imports it, not just this source file?  If yes, it seems like this may be breaking anything expecting e.g. a specific `html` library (or a specific version), like https://pypi.python.org/pypi/html

**Comment by olevold** [comment]
I still get  `ImportError: No module named 'ttypes'`

**Comment by MrSaints** [comment]
Are you running on Python 3 @olevold ? There's still work that needs to be done (e.g. on Thrift side) to get it working on Python 3.

**Comment by olevold** [comment]
Ok, I thought it had been solved. Sorry
